### ISSUE TITLE
[BUG] A function variable with nil value cannot be asserted with `BeNil()`

### ISSUE THREAD

**Issue by shafeeqes** [nunnatsa/ginkgolinter#171]
**Describe the bug**

I have a test:
```
		It("the test should pass", func() {
			fn := func(string) error { return nil }
			fn = nil

			Expect(fn).To(BeNil())
		})
```

This yields an error 
```
ginkgo-linter: wrong error assertion. Consider using `Expect(fn).To(Succeed())` instead (ginkgolinter)
			Expect(fn).To(BeNil())
```
in the linter. 
I agree `Succeed()` also matches nil values, but I would like `BeNil()` as it's more explicit.

**To Reproduce**
1. first step

**Expected behavior**
`BeNil` to be possible to be used with nil function variables.

**Environment:**
 - OS: [darwin/arm64]
 - Version [v0.18.0](https://github.com/nunnatsa/ginkgolinter/releases/tag/v0.18.0)

**Additional context**
Probably introduced with https://github.com/nunnatsa/ginkgolinter/pull/146

**Comment by nunnatsa** [comment]
Thanks @shafeeqes - great catch.

**Comment by nunnatsa** [comment]
version [v0.18.1](https://github.com/nunnatsa/ginkgolinter/releases/tag/v0.18.1) was released with a fix to this issue.

**Comment by shafeeqes** [comment]
Thanks @nunnatsa for the super fast response!

**Comment by shafeeqes** [comment]
@nunnatsa Would you also take care to update ginkgolinter in golangci-lint?

**Comment by nunnatsa** [comment]
> @nunnatsa Would you also take care to update ginkgolinter in golangci-lint?

No, sorry. They have their own automatic process for that. I guess it will take a few more days.
